### PR TITLE
Adds C# Documentation

### PR DESCRIPTION
**Pull request by acupofjose**
## What kind of change does this PR introduce?

Adds c# documentation to the website in to the latest v0.6 of supabase-csharp

## What is the current behavior?

There is no documentation for the c# library.

## Additional context

The codeblocks all say "c#" at the top of them instead of disappearing for syntax highlighting. This is probably a config issue on my part - any suggestions would be appreciated!

### CODE REVIEW

**Comment by acupofjose** [comment]
Hey team, could I have someone review this PR?

**Comment by dannykng** [comment]
> Hey team, could I have someone review this PR? Maybe dannykng?

Hey @acupofjose this looks awesome! The docs built successfully and you can view the preview here: https://docs-5ngj76gva-supabase.vercel.app/docs/reference/csharp/introduction

Going to loop in a couple engineers to review—thanks for adding this!

**Comment by acupofjose** [comment]
Appreciate that @dannykng!

**Comment by dannykng** [comment]
> The codeblocks all say "c#" at the top of them instead of disappearing for syntax highlighting. This is probably a config issue on my part - any suggestions would be appreciated!

I think [https://github.com/supabase/supabase/blob/master/apps/docs/components/reference/RefFunctionSection.tsx](https://github.com/supabase/supabase/blob/master/apps/docs/components/reference/RefFunctionSection.tsx) just needs to be updated for C# like https://github.com/supabase/supabase/pull/11235/files#diff-f8abb043eb41127bcf5d3ad2293f4ae975e85a88cf4e8261488e021306c0b262

**Comment by acupofjose** [comment]
@MildTomato could you take a look at this for me?

**Comment by saltcod** [comment]
Thanks @acupofjose !  Looks great. Going through this now. 

Could you add a light icon as well? I've made one here:
https://gist.githubusercontent.com/saltcod/165bfaa4808b0f8be0c4a1da1f90e4d9/raw/6f5ad99428b8634f08439c7a28f0af2c13c91bc4/reference-csharp-light.svg

File: spec/supabase_csharp_v0.yml [hunk 1]
```diff
@@ -0,0 +1,1669 @@
...
+            .Select("name, country_id")
+            .Single();
+          ```
+
+  - id: using-filters
+    title: Using Filters
+    description: |
+      Filters allow you to only return rows that match certain conditions.
+
+      Filters can be used on `Select()`, `update()`, and `Delete()` queries.
```
Comment (saltcod) [inline spec/supabase_csharp_v0.yml:1139]: should update() be capitalized?

File: spec/supabase_csharp_v0.yml [hunk 2]
```diff
@@ -0,0 +1,1669 @@
...
+  #                 // Return your widget with the data from the snapshot
+  #               },
+  #             );
+  #           }
+  #         }
+  #         ```
+
+  - id: list-buckets
+    description: |
+      Retrieves the details of all Storage buckets within an existing product.
```
Comment (saltcod) [inline spec/supabase_csharp_v0.yml:768]: You'll notice that "buckets" doesn't get output properly here. That's due to separate issue that we've fixed yesterday. You should get that after you merge master in.

CleanShot 2023-01-24 at 12 34 49@2x

**Comment by acupofjose** [comment]
@saltcod thanks for the review! Yes - I've added/fixed your requested changes.

**Comment by saltcod** [comment]
Sweeeet! I'll look at this first thing Monday morning! Thanks @acupofjose!

**Review by saltcod — APPROVED** [review]

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import spec_dart_v1 from '~/../../spec/supabase_dart_v1.yml' assert { type: 'yml
 // @ts-expect-error
 import spec_dart_v0 from '~/../../spec/supabase_dart_v0.yml' assert { type: 'yml' }
 // @ts-expect-error
+import spec_csharp_v0 from '~/../../spec/supabase_csharp_v0.yml' assert { type: 'yml' }
 import spec_python_v2 from '~/../../spec/supabase_py_v2.yml' assert { type: 'yml' }
```
Comment (saltcod) [inline apps/docs/components/Navigation/NavigationMenu/NavigationMenu.tsx:16]:
```suggestion
// @ts-expect-error
import spec_python_v2 from '~/../../spec/supabase_py_v2.yml' assert { type: 'yml' }
```

**Comment by saltcod** [comment]
Merged! Thanks @acupofjose — huge effort on this one. Greatly appreciated!

**Comment by acupofjose** [comment]
@saltcod thanks for the review/merge/effort! Exciting to see it merged.

**Pull request merged**